appserver/payara-appserver-modules/healthcheck-metrics/src/main/java/fish/payara/healthcheck/microprofile/metrics/MicroProfileMetricsCheck.java

```diff
@@ -0,0 +1,251 @@
+import org.jvnet.hk2.annotations.Service;
+
+@Service(name = "healthcheck-mpmetrics")
+@RunLevel(StartupRunLevel.VAL)
+public class MicroProfileMetricsCheck
+        extends BaseHealthCheck<HealthCheckMicroProfileMetricstExecutionOptions, MicroProfileMetricsChecker> {
+
+    @PostConstruct
+    public void postConstruct() {
+        postConstruct(this, MicroProfileMetricsChecker.class);
+    }
+
+    @Override
+    public HealthCheckMicroProfileMetricstExecutionOptions constructOptions(MicroProfileMetricsChecker checker) {
+        return new HealthCheckMicroProfileMetricstExecutionOptions(Boolean.valueOf(
+                checker.getEnabled()),
+                Long.parseLong(checker.getTime()),
+                asTimeUnit(checker.getUnit()),
+                checker.getMetricsScope(),
+                checker.getMetricApplicationName(),
+                checker.getMetricName());
+    }
+
+    @Override
+    protected String getDescription() {
+        return "healthcheck.description.MPmetrics";
+    }
+
+    @Override
+    protected HealthCheckResult doCheckInternal() {
+        String registryName = options.getMetricsScope();
+        List<String> metricNames = new ArrayList<>(Arrays.asList(options.getMetricName().split(",")));
+        metricNames.removeAll(Arrays.asList("", null));
+        MetricsService metricsService = Globals.getDefaultBaseServiceLocator().getService(MetricsService.class);
+        if (registryName.equalsIgnoreCase("application")) {
+            registryName = options.getMetricsApplicationName();
+        }
+
+        HealthCheckResult result = new HealthCheckResult();
+        List<String> metrics = collectMetrics(metricsService.getRegistry(registryName), metricNames);
+
+        result.add(new HealthCheckResultEntry(metrics.isEmpty() ? WARNING : GOOD,
+                metrics.isEmpty() ? "The metric you entered doesn't exist under " + registryName : metrics.stream().map(Object::toString).collect(Collectors.joining())));
+
+        return result;
+
+    }
+
+    private List<String> collectMetrics(MetricRegistry state, List<String> metricNames) {
+        List<String> array = new ArrayList<>();
+        String metricsInfos;
+        if (metricNames == null || metricNames.isEmpty()) {
+            for (String name : state.getNames()) {
+                metricsInfos = getMetricInfos(name, state);
+                if (metricsInfos != null) {
+                    array.add(metricsInfos);
+                }
+
+            }
+        } else {
+            for (String metricName : metricNames) {
+                metricsInfos = getMetricInfos(metricName.trim(), state);
+                if (metricsInfos != null) {
+                    array.add(metricsInfos);
+                }
+            }
+        }
+        return array;
+    }
+
+    private String getMetricInfos(String metricName, MetricRegistry state) {
+        Map<MetricID, Metric> metricInfos = state.getMetrics().entrySet().stream()
```

Comment:
Does this need repeating each loop? Also, users won't be familiar with the dot notation of the metrics - they're mote likely to use underscore notation as it appears in the metrics result